WeatherForecast/WeatherForecast/SceneDelegate.swift

```diff
@@ -15,7 +15,21 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
         // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
         // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
-        guard let _ = (scene as? UIWindowScene) else { return }
+        guard let scene = (scene as? UIWindowScene) else { return }
+        
+        let weatherListViewController: WeatherListViewController = .init(
+            weatherRepository: WeatherRepositoryImpl(dataSource: WeatherDataSourceImpl(), decoder: DataDecoderJSON(decoder: JSONDecoder())),
+            imageFetcher: ImageFetcherImpl()
+        )
+        let navigationController: UINavigationController = UINavigationController(rootViewController: weatherListViewController)
+        navigationController.navigationBar.prefersLargeTitles = true
+        navigationController.navigationBar.tintColor = .black
+        
+        let window: UIWindow = UIWindow(windowScene: scene)
+        window.rootViewController = navigationController
```

Comment:
```suggestion
     struct AppDependency {
  let window: UIWindow
}

extension AppDependency {
  static func resolve(_ scene: UIWindowScene) -> AppDependency {
    let window: UIWindow = UIWindow(windowScene: scene)
    window.makeKeyAndVisible()
    // 필요한 객체 생성
    }
    
    let viewController = ViewController(
      // initialize
    )
    
    let navigationController: UINavigationController = UINavigationController(rootViewController: viewController)
    
    window.backgroundColor = .white
    window.rootViewController = navigationController
    
    return AppDependency(
      window: window
    )
  }
}
```

이러한 방식으로 AppDependency를 다른 파일로 분리해서